sklearn/decomposition/_dict_learning.py

```diff
@@ -1043,6 +1043,10 @@ def fit(self, X, y=None):
         """
         return self
 
+    @property
+    def n_features_in_(self):
+        return self.components_.shape[1]
```

Comment:
check if fitted like the one in `_target.py`?

we could also have a decorator for this though.

Reply:
No need, `self.components_` exists right after init is called. It's in the SLEP.